### PR TITLE
dep: update tidb to fix memory leak

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/2124

### What is changed and how it works?
update TiDB to introduce https://github.com/pingcap/tidb/pull/28027

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: dm/ctl/common/util.go [hunk 1]
```diff
@@ -25,7 +25,9 @@ import (
 	"sync"
 	"time"
 
+	// nolint:staticcheck
```
Comment (GMHDBJD) [inline dm/ctl/common/util.go:28]: How about add `exclude-rules` like #2061

File: pkg/v1workermeta/meta.go [hunk 2]
```diff
@@ -14,6 +14,7 @@
 package v1workermeta
 
 import (
+	// nolint:staticcheck
```
Comment (okJiang) [inline pkg/v1workermeta/meta.go:17]: What does it do?

**Comment by lance6716** [comment]
waiting https://github.com/PingCAP-QE/go-sqlsmith/pull/13

**Comment by lance6716** [comment]
waiting https://github.com/pingcap/dumpling/pull/352

**Comment by Ehco1996** [comment]
```
] /tmp/dm_test/all_mode/worker1/log/stdout.log

[2021-09-15T01:00:55.481Z] + cat /tmp/dm_test/all_mode/worker1/log/stdout.log

[2021-09-15T01:00:55.481Z] panic: proto: message backup.BackupMeta is already registered

[2021-09-15T01:00:55.481Z] See https://developers.google.com/protocol-buffers/docs/reference/go/faq#namespace-conflict
```

seems pb have conflicts 😂

**Review by glorv — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**